thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/ThirdEyeCacheResponse.java

```diff
@@ -127,26 +127,29 @@ public boolean isMissingEndSlice(long sliceEnd) {
    */
   private void checkAndLogMissingMiddleSlices() {
 
-    List<String> missingTimestamps = new ArrayList<>();
-    long timeGranularity = request.getRequest().getGroupByTimeGranularity().toMillis();
-
-    // remember that we return the cached timeseries in sorted order,
-    // but this assumption is not necessarily true if mergeSliceIntoRows() has been called.
-    for (int i = 1; i < timeSeriesRows.size(); i++) {
-      long previousTimestamp = timeSeriesRows.get(i - 1).getTimestamp();
-      long currentTimestamp = timeSeriesRows.get(i).getTimestamp();
-
-      // add all missing timestamps between previous timestamp and current timestamp to
-      // the list of missing timestamps.
-      while (previousTimestamp + timeGranularity < currentTimestamp) {
-        missingTimestamps.add(String.valueOf(previousTimestamp + timeGranularity));
-        previousTimestamp += timeGranularity;
+    if (!this.hasNoRows()) {
+      List<String> missingTimestamps = new ArrayList<>();
+      long timeGranularity = request.getRequest().getGroupByTimeGranularity().toMillis();
+
+      // remember that we return the cached timeseries in sorted order,
+      // but this assumption is not necessarily true if mergeSliceIntoRows() has been called.
+      for (int i = 1; i < timeSeriesRows.size(); i++) {
+        long previousTimestamp = timeSeriesRows.get(i - 1).getTimestamp();
+        long currentTimestamp = timeSeriesRows.get(i).getTimestamp();
+
+        // add all missing timestamps between previous timestamp and current timestamp to
+        // the list of missing timestamps.
+        while (previousTimestamp + timeGranularity < currentTimestamp) {
+          missingTimestamps.add(String.valueOf(previousTimestamp + timeGranularity));
+          previousTimestamp += timeGranularity;
+        }
       }
-    }
 
-    if (missingTimestamps.size() > 0) {
-      LOG.info("cached time-series for metricUrn {} was missing data points in the middle for {} timestamps: {}",
-          request.getMetricUrn(), missingTimestamps.size(), String.join(",", missingTimestamps));
+      // we will need to evaluate whether this is generating too many logs.
+      if (missingTimestamps.size() > 0) {
+        LOG.info("cached time-series for metricUrn {} was missing data points in the middle for {} timestamps: {}",
```

Comment:
Instead of logging all the missing timestamps, just log the start and end timestamp along with the granularity (in days).

 - TimeUnit.MILLISECONDS.toDays(granularity)

Reply:
There can be multiple holes in the middle, so just logging start and end timestamps isn't necessarily enough. Logging the intervals (e.g. start1-end1, start2-end2, start3-end3, ...) of all the missing slices may be better.

Reply:
added